azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java

```diff
@@ -117,144 +118,95 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
+                        public Single<HttpResponse> call(PollStrategy pollStrategy) {
+                            return pollStrategy.pollUntilDone();
                         }
                     });
-            result = super.handleAsyncHttpResponse(httpRequest, asyncHttpResponse, methodParser, returnType);
+            result = handleAsyncHttpResponseInner(httpRequest, lastAsyncHttpResponse, methodParser, returnType);
         }
 
         return result;
     }
 
-    private static PollStrategy createPollStrategy(HttpRequest httpRequest, HttpResponse httpResponse, SerializerAdapter<?> serializer) {
-        PollStrategy result = null;
-        final long delayInMilliseconds = defaultDelayInMilliseconds();
+    private Single<PollStrategy> createPollStrategy(final HttpRequest originalHttpRequest, final Single<HttpResponse> asyncOriginalHttpResponse, final SerializerAdapter<?> serializer) {
+        return asyncOriginalHttpResponse
+                .flatMap(new Func1<HttpResponse, Single<PollStrategy>>() {
+                    @Override
+                    public Single<PollStrategy> call(final HttpResponse originalHttpResponse) {
+                        Single<PollStrategy> result = null;
+                        final long delayInMilliseconds = defaultDelayInMilliseconds();
+
+                        final int httpStatusCode = originalHttpResponse.statusCode();
+                        if (httpStatusCode != 200) {
+                            PollStrategy pollStrategy = null;
+                            final String fullyQualifiedMethodName = originalHttpRequest.callerMethod();
+                            final String originalHttpRequestMethod = originalHttpRequest.httpMethod();
+                            final String originalHttpRequestUrl = originalHttpRequest.url();
+
+                            if (originalHttpRequestMethod.equalsIgnoreCase("PUT") || originalHttpRequestMethod.equalsIgnoreCase("PATCH")) {
+                                if (httpStatusCode == 201) {
+                                    pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, fullyQualifiedMethodName, originalHttpResponse, originalHttpRequestUrl, serializer, delayInMilliseconds);
+                                } else if (httpStatusCode == 202) {
+                                    pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, fullyQualifiedMethodName, originalHttpResponse, originalHttpRequestUrl, serializer, delayInMilliseconds);
+                                    if (pollStrategy == null) {
+                                        pollStrategy = LocationPollStrategy.tryToCreate(AzureProxy.this, fullyQualifiedMethodName, originalHttpResponse, delayInMilliseconds);
+                                    }
+                                }
+                            } else /* if (originalRequestHttpMethod.equalsIgnoreCase("DELETE") || originalRequestHttpMethod.equalsIgnoreCase("POST") */ {
```

Comment:
Is this dead code in the comment? Any ambiguity about whether that condition is needed?

Reply:
I originally had it in to make it clear what the else condition was, but I'll take it out.